airbyte-integrations/connectors/source-salesforce/source_salesforce/source.py

```diff
@@ -96,6 +104,14 @@ def read(
                     connector_state=connector_state,
                     internal_config=internal_config,
                 )
+            except exceptions.HTTPError as error:
+                error_data = error.response.json()[0]
+                error_code = error_data.get("errorCode")
+                if error.response.status_code == codes.FORBIDDEN and error_code == "REQUEST_LIMIT_EXCEEDED":
+                    logger.warn(f"API Call limit is exceeded. Error message: '{error_data.get('message')}'")
+                    break  # if got 403 rate limit response, finish the sync with success.
```

Comment:
Connector should stop the sync if one stream reached rate limit
If the connector has stream_1, stream_2, stream_3.
While reading `stream_1` if 403 (Rate Limit) is received, it should finish that stream with success and stop the sync process.
The following streams should not be executed.